compiler/rustc_hir_analysis/src/impl_wf_check.rs

```diff
@@ -181,6 +181,15 @@ fn report_unused_parameter(tcx: TyCtxt<'_>, span: Span, kind: &str, name: Symbol
         name
     );
     err.span_label(span, format!("unconstrained {} parameter", kind));
+    err.span_suggestion(
+        span,
+        format!(
+            "Either remove the type parameter '{}' or make use of it, for example ` for S<{}>`.",
+            name, name
+        ),
+        "",
```

Comment:
I don't think it's a good idea to give a suggestion to remove it. Simply removing the parameter is usually not the solution to this error in my experience.

Reply:
I based myself on the "desired output" on https://github.com/rust-lang/rust/issues/107295
What would be a good explanation? :)